File: app/src/main/java/vasuki/istanpdf/domain/ImagesToPdf.java
package vasuki.istanpdf.domain;

import android.net.Uri;

import java.util.List;

import vasuki.istanpdf.data.PdfEngine;
import vasuki.istanpdf.model.PageItem;

public final class ImagesToPdf {
    private final PdfEngine pdfEngine;

    public ImagesToPdf(PdfEngine pdfEngine) {
        this.pdfEngine = pdfEngine;
    }

    public void execute(List<Uri> sources, List<PageItem> pages, Uri destination) throws Exception {
        if (sources == null || sources.isEmpty()) {
            throw new IllegalArgumentException("At least one image is required");
        }
        pdfEngine.imagesToPdf(sources, pages, destination);
    }
}

File: app/src/main/kotlin/vasuki/istanpdf/domain/ImagesToPdf.kt
package vasuki.istanpdf.domain

import android.net.Uri
import vasuki.istanpdf.data.PdfEngine
import vasuki.istanpdf.model.PageItem

class ImagesToPdf(private val pdfEngine: PdfEngine) {
    @Throws(Exception::class)
    fun execute(sources: List<Uri>, pages: List<PageItem>, destination: Uri) {
        require(sources.isNotEmpty()) { "At least one image is required" }
        pdfEngine.imagesToPdf(sources, pages, destination)
    }
}
